lib/system.nim

```diff
@@ -2668,7 +2668,8 @@ when not defined(nimscript) and hasAlloc:
       ""
 
 template accumulateResult*(iter: untyped) =
-  ## helps to convert an iterator to a proc.
+  ## helps to convert an iterator to a proc. 
+  ## See also `sequtils.toSeq` which is more hygienic and efficient.
```

Comment:
This is only less efficient when iter.len compiles, in any other case it's a very very small advantage over toSeq. Another place where toSeq is weak however is things like strutils.split, since split(str).len compiles but only because split(str) has both a proc and iterator definiton where the proc uses this template to return a seq that fits the condition. In that case, toSeq compiles to:

```nim
var i = 0
var result = newSeq[string](split(str).len) # split(str) is iterated through for the first time
for x in split(str): # and the second time
  result[i] = x
  inc i
result
```

The root cause for this being toSeq's one weak spot is that this template exists and encourages iterator masking like what split does. I vote to remove this template altogether and all... 8 usages of it (all in string modules) in the standard library.

Reply:
Yeah, seems like a candidate for deprecation.

Reply:
Yeah, let's deprecate it.

Reply:
PTAL: I've pushed a new commit showing deprecation